fix(template): dynamically set default schema alias in typescript type generation

bug-author opened the pull request:
## What kind of change does this PR introduce?

This PR fixes the incorrect default alias generation for TypeScript type schemas when using a single non‑public schema with `supabase gen types`.

## What is the current behavior?

Currently, the TypeScript template always hardcodes the default schema alias as:

```ts
export type PublicSchema = Database[Extract<keyof Database, "public">]
```

This means that when a user generates types using a non‑public schema (for example, `--schema robo`), the generated helper types (for example. `Tables<"test_table">`) incorrectly reference the public schema, even if no tables exist there.

Relevant issues: 
- cli:  [#3146](https://github.com/supabase/cli/issues/3146), [#3061](https://github.com/supabase/cli/issues/3061)
- postgres-meta: #833 (for additional context)

## What is the new behavior?

This PR updates the TypeScript template in pg‑meta so that when exactly one non‑public schema is provided, the template automatically uses the environment variable `PG_META_GENERATE_TYPES_DEFAULT_SCHEMA` to generate a default alias. For example, if the schema provided is "robo", the template will now output:

```ts
export type DefaultSchema = Database[Extract<keyof Database, "robo">]
export type DefaultSchemaOrPublic = DefaultSchema
```

The helper types (`Tables`, `TablesInsert`, etc.) reference `DefaultSchemaOrPublic` instead of the hardcoded "public". In multi‑schema scenarios, the environment variable is not set, and the default behavior (using "public") remains, the users can explicitly specify the schema for non-public tables.

This change ensures:

- In single‑schema mode with a non‑public schema, shorthand type references (like `Tables<"test_table">`) correctly resolve to `Database["robo"]["Tables"]["test_table"]`.
- In multi‑schema mode, users must explicitly specify which schema to use (for example, `Tables<{ schema: "robo" }, "test_table">`).

## Additional context
Usage example

```
CREATE SCHEMA robo;
CREATE TABLE robo.test_table (
  id SERIAL PRIMARY KEY,
  name TEXT
);


CREATE TABLE public.example_table (
  id SERIAL PRIMARY KEY,
  description TEXT
);
```

```ts
// Scenario 1: No schema flag (defaults to "public")
import * as PublicTypes from "./dbTypes-public";

type PublicExampleRow = PublicTypes.Tables<"example_table">;

// Scenario 2: Single non‑public schema (e.g. "robo")
import * as RoboTypes from "./dbTypesRobo";

type RoboExampleRow = RoboTypes.Tables<"test_table">;

// Scenario 3: Multiple schemas specified (e.g. "public" and "robo")
import * as MultiTypes from "./dbTypes-multi";

type MultiPublicRow = MultiTypes.Tables<"example_table">;
type MultiRoboRow = MultiTypes.Tables<{ schema: "robo" }, "test_table">;

// examples
const publicRow: PublicExampleRow = {
  id: 1,
  description: "A public record",
};


const roboRow: RoboExampleRow = {
  id: 10,
  name: "Robo record",
};

const multiPublicRow: MultiPublicRow = {
  id: 2,
  description: "Multi public record",
};

const multiRoboRow: MultiRoboRow = {
  id: 20,
  name: null,
};
```

Corresponding [cli PR](https://github.com/supabase/cli/pull/3243)

soedirgo commented on src/server/templates/typescript.ts:
```diff
@@ -9,6 +9,8 @@ import type {
 } from '../../lib/index.js'
 import type { GeneratorMetadata } from '../../lib/generators.js'
 
+const defaultSchema = process.env.PG_META_GENERATE_TYPES_DEFAULT_SCHEMA
```
This should be in `src/server/constants.ts`

bug-author replied:
Fixed

soedirgo commented on src/server/templates/typescript.ts:
```diff
@@ -390,11 +392,18 @@ export type Database = {
-type PublicSchema = Database[Extract<keyof Database, "public">]
+${
+  defaultSchema
+    ? `export type DefaultSchema = Database[Extract<keyof Database, ${JSON.stringify(defaultSchema)}>]
+  export type DefaultSchemaOrPublic = DefaultSchema`
+    : `export type PublicSchema = Database[Extract<keyof Database, "public">]
+  export type DefaultSchemaOrPublic = PublicSchema`
```
This should be kept private and unexported

bug-author replied:
This is now refactored to 
```ts
type DefaultSchema = Database[Extract<keyof Database, ${JSON.stringify(DEFAULT_SCHEMA) || '"public"'}>]
```

soedirgo commented on src/server/templates/typescript.ts:
```diff
@@ -390,11 +392,18 @@ export type Database = {
+${
+  defaultSchema
+    ? `export type DefaultSchema = Database[Extract<keyof Database, ${JSON.stringify(defaultSchema)}>]
+  export type DefaultSchemaOrPublic = DefaultSchema`
+    : `export type PublicSchema = Database[Extract<keyof Database, "public">]
+  export type DefaultSchemaOrPublic = PublicSchema`
+}
+
 
 export type Tables<
   PublicTableNameOrOptions extends
-    | keyof (PublicSchema["Tables"] & PublicSchema["Views"])
+    | keyof (DefaultSchemaOrPublic["Tables"] & DefaultSchemaOrPublic["Views"])
```
Better to use `DefaultSchema` nomenclature everywhere vs. `PublicSchema`

bug-author replied:
Resolved

soedirgo commented:
Thanks @bug-author! Left some comments

bug-author commented:
@soedirgo Thank you for the review. I have made the suggested changes:

- Moved the default schema constant to `src/server/constants.ts`
- Refactored schema aliases to use `DefaultSchema` only

Thanks for helping me refine the implementation.

soedirgo reviewed (approved):
Thanks again @bug-author!
